### PR TITLE
PYIC-7650: Add M1C matching API tests

### PR DESCRIPTION
**Pull request by MikeCollingwood**
### What changed

- Add M1C matching API tests
- See https://govukverify.atlassian.net/browse/PYIC-7650?focusedCommentId=191131

### Why did it change

- Coverage of cases, one of which was picked up in QA

### Issue tracking

- [PYIC-7650](https://govukverify.atlassian.net/browse/PYIC-7650)

[PYIC-7650]: https://govukverify.atlassian.net/browse/PYIC-7650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature [hunk 1]
```diff
@@ -1,138 +1,214 @@
...
+  Rule: Match M1C
+
+    Background:
+      Given the subject already has the following credentials
+        | CRI     | scenario               |
+        | dcmaw   | kenneth-passport-valid |
+        | address | kenneth-current        |
+      And the subject already has the following expired credentials
+        | CRI   | scenario        |
+        | fraud | kenneth-score-2 |
```
Comment (DanCorderIPV) [inline api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature]: It doesn't make any difference to whether the test passes, but to be realistic I think this should also be `kenneth-no-applicable`
People should only be getting this response for international addresses and we aren't changing the address in all the test that use this.

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature [hunk 2]
```diff
@@ -1,138 +1,214 @@
...
+      When I submit 'kenneth-changed' details to the CRI stub
+      Then I get a 'fraud' CRI response
+      When I submit 'kenneth-no-applicable' details to the CRI stub
+      Then I get a 'page-ipv-success' page response with context 'updateIdentity'
+      When I submit a 'next' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P2' identity
+
+    @MikeTest
```
Comment (DanCorderIPV) [inline api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature]: Oops :)

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature (hunk 1 above)
Comment (DanCorderIPV) [reply api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature]: Could also have tests for users changing to an international address I guess, but that's probably overkill

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature [hunk 3]
```diff
@@ -1,138 +1,214 @@
...
-    Given the subject already has the following credentials
-      | CRI     | scenario                     |
-      | dcmaw   | kenneth-driving-permit-valid |
-      | address | kenneth-current              |
-    And the subject already has the following expired credentials
-      | CRI   | scenario        |
-      | fraud | kenneth-score-2 |
-    When I start a new 'medium-confidence' journey
-    Then I get a 'confirm-your-details' page response
+  Rule:
```
Comment (DanCorderIPV) [inline api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature]: Is it OK to leave this rule blank?

**Review by DanCorderIPV — CHANGES_REQUESTED** [review]

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature (hunk 1 above)
Comment (MikeCollingwood) [reply api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature]: Oh I see your point, in some cases they could have moved abroad but really the tests don't always have changed address.

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature (hunk 3 above)
Comment (MikeCollingwood) [reply api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature]: It seems the other scenarios only match M1B, so I may use that to name the rule

Comment (MikeCollingwood) [reply api-tests/features/repeat-fraud-check/p2-repeat-fraud-check.feature:4]: I suppose we leave Background blank

**Review by DanCorderIPV — APPROVED** [review]

**Comment by DanCorderIPV** [comment]
I think it's safe to merge this today as it's only tests

**Pull request merged**